Duplicati/Library/Backend/Sia/Sia.cs

```diff
@@ -0,0 +1,433 @@
+            }
+            return fl;
+        }
+
+        private bool IsDownloadComplete(string siafilename, string localname)
+        {
+            SiaDownloadList fl = GetDownloads();
+            foreach (var f in fl.Files)
+            {
+                if (f.Siapath == siafilename)
+                {
+                    if (f.Error != "")
+                    {
+                        throw new Exception("failed to download " + siafilename + "err: " + f.Error);
+                    }
+                    if (f.Filesize == f.Received)
+                    {
+                        try
+                        {
+                            // Sia seems to keep the file open/locked for a while, make sure we can open it
+                            System.IO.FileStream fs = new System.IO.FileStream(localname, System.IO.FileMode.Open, System.IO.FileAccess.Read, System.IO.FileShare.ReadWrite);
+                            fs.Close();
+                        }
+                        catch (System.IO.IOException)
+                        {
+                            return false;
+                        }
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        #region IBackend Members
+
+        public void Test()
+        {
+            List();
+        }
+
+        public void CreateFolder()
+        {
+            // Dummy method, Sia doesn't have folders
+        }
+
+        public string DisplayName
+        {
+            get { return Strings.Sia.DisplayName; }
+        }
+
+        public string ProtocolKey
+        {
+            get { return "sia"; }
+        }
+
+         
+        public List<IFileEntry> List()
+        {
+            var files = new List<IFileEntry>();
+            try
+            {
+                SiaFileList fl = GetFiles();
+                foreach (var f in fl.Files)
+                {
+                    // Sia returns a complete file list, but we're only interested in files that are
+                    // in our target path
+                    if (f.Siapath.StartsWith(m_targetpath))
+                    {
+                        FileEntry fe = new FileEntry(f.Siapath.Substring(m_targetpath.Length+1));
+                        fe.Size = f.Filesize;
+                        fe.IsFolder = false; 
+                        files.Add(fe);
+                    }
+                }
+            }
+            catch (System.Net.WebException wex)
+            {
+                throw new Exception("failed to call /renter/files "+wex.Message);
+            }
+
+            return files;
+        }
+
+        public void Put(string remotename, string filename)
+        {
+            string endpoint ="";
+            string siafile = m_targetpath + "/" + remotename;
+
+            try {
+                endpoint = string.Format("/renter/upload/{0}/{1}?source={2}&datapieces=1&paritypieces=12",
```

Comment:
why are the redundancy settings being specified here? If unspecified, the recommended settings are used.

Reply:
Oops.  That's some leftovers, I was thinking they should be configurable options but didn't get that far.